syft/frameworks/torch/tensors/interpreters/replicated_shared.py

```diff
@@ -270,6 +281,128 @@ def shape(self):
     def players(self):
         return self.__get_players()
 
+    def set_garbage_collect_data(self, value):
+        shares = self.child
+
+        for _, shares in shares.items():
+            for share in shares:
+                assert share.is_wrapper is True
+                share.gc = value
```

Comment:
why do you set `gc` and not `.garbage_collect_data` ?

Reply:
This is because ```share``` is a wrapper.
And in the ```native``` we have:
```
@gc.setter
def gc(self, flag):
     self.garbage_collection = flag
...
@garbage_collection.setter
def garbage_collection(self, flag):
      if not self.has_child():
          if hasattr(self, "ptr") and self.ptr is not None:
              self.child = self.ptr
      self.child.garbage_collect_data = flag
```
Basically we would do. ```share.child.garbage_collect_data = flag```

Reply:
Ah ok that's nice! Thanks :)